include/openssl/arm_arch.h

```diff
@@ -117,8 +75,28 @@
 // ARMV8_PMULL indicates support for carryless multiplication.
 #define ARMV8_PMULL (1 << 5)
 
+// ARMV8_SHA512 indicates support for hardware SHA-512 instructions.
+#define ARMV8_SHA512 (1 << 6)
+
 #if defined(__ASSEMBLER__)
 
+// We require the ARM assembler provide |__ARM_ARCH| from Arm C Language
+// Extensions (ACLE). This is supported in GCC 4.8+ and Clang 3.2+. MSVC does
```

Comment:
is this going to be a problem for us? do we use MSVC in the CI or clang? (and what about the infamous gcc4.1?

Reply:
MSVC is used in aws-lc-ci-windows-x86.